pkg/controller/gkenetworkparamset/gnpcontroller_validations.go

```diff
@@ -206,7 +206,7 @@ func (val *gnpNetworkCrossValidation) toCondition() v1.Condition {
 }
 
 // crossValidateNetworkAndGnp validates a given network and GNP object are compatible
-func crossValidateNetworkAndGnp(network *networkv1.Network, params *networkv1.GKENetworkParamSet) *gnpNetworkCrossValidation {
+func crossValidateNetworkAndGnp(network *networkv1.Network, params *networkv1.GKENetworkParamSet, originalParams *networkv1.GKENetworkParamSet) *gnpNetworkCrossValidation {
```

Comment:
We really shouldn't change the function signature here. This is a validation function. A good design pattern is to keep validation functions stateless. i.e. given some input, it should validate if the input is consistent with the rest of the world as it exists today.

Reply:
Done, remove originalParams